bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/OpenBuilder.java

```diff
@@ -0,0 +1,83 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.common.concurrent.FutureUtils;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+
+/**
+ * Builder-style interface to open exiting ledgers.
+ *
+ * @since 4.6
+ */
+public interface OpenBuilder extends OpBuilder<ReadHandle> {
+
+    /**
+     * Set the id of the ledger to be opened.
+     *
+     * @param ledgerId
+     *
+     * @return the builder itself
+     */
+    OpenBuilder withLedgerId(long ledgerId);
+
+    /**
+     * Define to open the ledger in recovery mode or in readonly mode. In recovery mode the ledger will be fenced and
+     * the writer of the ledger will be prevented from issuing other writes to the ledger. It defaults to 'true'
```

Comment:
Can you change the default to 'false'? I think in general people would like to open ledger to read.

Reply:
done